### PR TITLE
deployment changes

### PR DESCRIPTION
**Pull request by sthombare**
Changed pubsub and sample publisher/subscriber deployment files to fill namespace name from env variable. Also, added node selector.

### CODE REVIEW

File: deploy/pubsub-pub.yaml [hunk 1]
```diff
@@ -19,6 +19,9 @@ spec:
           - name: grpc
             containerPort: 80
         args: [
-            "-url=pubsub.atlas.svc:80",
+            "-url=pubsub.$PUBSUB_NAMESPACE.svc:80",
```
Comment (jhall38) [inline deploy/pubsub-pub.yaml]: Should we switch to a different default port for ClusterIP service? 80 seems like a strange choice.

**Comment by jhall38** [comment]
Will we eventually move these files to saas-app-deployment?

File: deploy/pubsub-pub.yaml (hunk 1 above)
Comment (dmacthedestroyer) [reply deploy/pubsub-pub.yaml]: what's strange about port 80?

Comment (sthombare) [reply deploy/pubsub-pub.yaml]: port 80 is generally used for web client to receive HTML page. perhaps this makes it unconventional choice. But not an issue.

**Comment by sthombare** [comment]
@dmacthedestroyer 

Are you referring to following change in make target or something else that I am missing? I thought the rule won't work as pubsub.yaml is not in same directory as makefile.  One would have to call make -f ../Makefile temp-create-pubsub  which I wasn't sure was intended. If yes, I will add that.

-       cat pubsub.yaml | envsubst ' $AWS_ACCESS_KEY$AWS_ACCESS_KEY_ID$AWS_SECRET_ACCESS_KEY$AWS_SECRET_KEY'| ks create -f pubsub.yaml
+       cat pubsub.yaml | envsubst ' $AWS_ACCESS_KEY$AWS_ACCESS_KEY_ID$AWS_SECRET_ACCESS_KEY$AWS_SECRET_KEY$PUBSUB_NAMESPACE'| ks create -f pubsub.yaml

**Comment by dmacthedestroyer** [comment]
that make target is indeed broken, and should not have been added to the Makefile in the first place. I was more referring to `make deploy*` and `make remove*`. 

As mentioned above, I'm on board with all of these k8s files no longer living in this repository. But I think there are teams that use this particular deployment file and the associated `make` targets to do local development on minikube. Updating the `Makefile` targets will ensure that it will still work for them after this change.

**Comment by sthombare** [comment]
@dmacthedestroyer -  updated deployment make targets. Please review.

**Pull request merged**